File: framework/jetpack/room/room/src/org/yuyan/room/entity/ColumnInfo.java
package org.yuyan.room.entity;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

/**
 * @param name
 * it's interfacing to a default rule
 * that we indicate a SQL field to be named with
 * "database name" + "_" + "field name"
 * etc: user_name, user_mail, ...
 * */
@Target({ElementType.FIELD, ElementType.METHOD})
public @interface ColumnInfo {
    String name();
}

File: bak/room/framework/enty/ColumnInfo.kt
package room.framework.enty

/**
 * @param name
 * it's interfacing to a default rule
 * that we indicate a SQL field to be named with
 * "database name" + "_" + "field name"
 * etc: user_name, user_mail, ...
 * */
@Target(AnnotationTarget.PROPERTY)
annotation class ColumnInfo(val name: String)